webdriver/tests/interaction/element_click.py

```diff
@@ -0,0 +1,214 @@
+    assert_error(response, "element not interactable")
+
+
+@pytest.mark.parametrize("element", ["""<input type=button value=Roger style="position: absolute; left: 10px; top: 10px">
+                                    <div style="position: absolute; height: 100px; width: 100px; background: rgba(255,0,0,.5); left: 10px; top: 5px"></div>""",
+                                    "<input type=button value=Roger style=\"pointer-events: none\">",
+                                    """<div style="position: absolute; height: 50px; width: 100px; background: rgba(255,0,0,.5); left: 10px; top: 50px; overflow: hidden">
+                                        ABCDEFGHIJKLMNOPQRSTUVWXYZ
+                                        <input type=text value=Federer style="position: absolute; top: 50px; left: 10px;">
+                                    </div>"""])
+
+def test_element_click_intercepted(session, element):
+    # 14.1 Step 6
+    # Test intercepted click
+    # Test no pointer events
+    # Test overflow hidden
+    session.url = inline(element)
+
+    element = session.find.css("input", all=False)
+    response = click(session, element)
+    assert_error(response, "element not interactable")
+
+
+def test_match_option_element(session, mouse_chain):
+    # 14.1 Step 7
+    session.url = inline("""<select id=sel style="position: absolute; left: 10px; top: 50px">
+                            <option id=op1></option>
+                            <option id=op2>Roger</option>
+                            <option id=op3>Federer</option>
+                        </select>
+
+                        <input id=focusCheck type=checkbox>
+                        <input id=mouseOverCheck type=checkbox>
+                        <input id=mouseMoveCheck type=checkbox>
+                        <input id=mouseDownCheck type=checkbox>
+                        <input id=mouseUpCheck type=checkbox>
+                        <input id=mouseClickCheck type=checkbox>
+                        <input id=inputEventCheck type=checkbox>
+                        <input id=changeEventCheck type=checkbox>
+                        <script>
+                        document.getElementById("sel").addEventListener("focus", checkFocus);
+                        document.getElementById("sel").addEventListener("mouseover", checkMouseOver);
+                        document.getElementById("sel").addEventListener("mousedown", checkMouseDown);
+                        document.getElementById("sel").addEventListener("mouseup", checkMouseUp);
+                        document.getElementById("sel").addEventListener("click", checkMouseClick);
+                        document.getElementById("sel").addEventListener("input", checkInputEvent);
+                        document.getElementById("sel").addEventListener("change", checkChangeEvent);
+
+                        window.addEventListener("mousemove", checkMouseMove);
+
+                        function checkFocus() {
```

Comment:
Instead of writing out these functions, can you use arrow pointer functions when calling `("sel").addEventListener` instead?

Reply:
fixed